stdlib/http/src/main/ballerina/http/service_endpoint.bal

```diff
@@ -164,13 +230,183 @@ public type KeepAlive "AUTO"|"ALWAYS"|"NEVER";
 @final public KeepAlive KEEPALIVE_NEVER = "NEVER";
 
 function Listener::init (ServiceEndpointConfiguration c) {
+    match c.authProviders {
+        AuthProvider[] providers => addAuthFiltersForSecureListener(c, self.instanceId);
+        () => {}
+    }
     self.config = c;
     var err = self.initEndpoint();
     if (err != null) {
         throw err;
     }
 }
 
+# Add authn and authz filters
+#
+# + config - `ServiceEndpointConfiguration` instance
```

Comment:
Shall we document instanceId??

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/11173/commits/34216c7f1de2b3ee7fd63331d5384c928a3fa0a7